tests/end2end/test_hints_html.py

```diff
@@ -146,3 +151,45 @@ def test_word_hints_issue1393(quteproc, tmpdir):
         quteproc.wait_for(message='hints: *', category='hints')
         quteproc.send_cmd(':follow-hint {}'.format(hint))
         quteproc.wait_for_load_finished('data/{}'.format(target))
+
+
+@pytest.mark.parametrize('min_len', [0, 3])
+@pytest.mark.parametrize('num_chars', [9])
+@pytest.mark.parametrize('num_elements', range(1, 26))
+def test_scattered_hints_count(win_registry, mode_manager, min_len,
+                               num_chars, num_elements):
+    """Test scattered hints function."""
+    # pylint: disable=W0141
+    manager = qutebrowser.browser.hints.HintManager(0, 0)
+    chars = string.ascii_lowercase[:num_chars]
+
+    hints = manager._hint_scattered(min_len, chars,
+                                    list(range(num_elements)))
+
+    # Check if hints are unique
+    assert len(hints) == len(set(hints))
+
+    # Check if any hints are shorter than min_len
+    assert not any([x for x in hints if len(x) < min_len])
+
+    # Check we don't have more than 2 link lengths
+    # Eg: 'a' 'bc' and 'def' cannot be in the same hint string
+    hint_lens = set(map(len, hints))
+    assert len(hint_lens) <= 2
+
+    if len(hint_lens) == 2:
+        # Check if hint_lens are more than 1 apart
+        # Eg: 'abc' and 'd' cannot be in the same hint sequence, but
+        # 'ab' and 'c' can
+        assert abs(functools.reduce(operator.sub, hint_lens)) <= 1
+
+    longest_hints = filter(lambda x: len(x) == max(hint_lens), hints)
+
+    if min_len < max(hint_lens) - 1:
+        # Check if we have any unique prefixes. For example, 'la'
+        # alone, with no other 'l<x>'
+        count_map = {}
+        for x in longest_hints:
+            prefix = x[:-1]
+            count_map[prefix] = count_map.get(prefix, 0) + 1
+        assert not any(filter(lambda x: x == 1, count_map.values()))
```

Comment:
I don't understand this - isn't this basically the equivalent of `any([1, 1, 1])` which will always be `True`, i.e. you're just using `any()` to see whether a list is empty?

This would be much clearer IMHO: `assert not any(e == 1 for e in count_map.values())`
or even better: `assert all(e != 1 for e in count_map.values())`